parser/src/ast/integrity_constraints.rs

```diff
@@ -41,6 +41,9 @@ pub enum IntegrityExpr {
     /// name of the matrix and indices of the element to access.
     MatrixAccess(MatrixAccess),
     Next(TraceAccess),
+    /// Represents a column in the execution trace, e.g. `$main[2]`. [ColumnAccess] contains the
+    /// segment index and column index to be accessed.
+    ColumnAccess(ColumnAccess),
```

Comment:
We now have 2 different ways of representing a trace access in the AST which are very different from each other. I'd prefer to have them be more similar.

I propose the following changes:
- rename the `TraceAccess` struct to `NamedTraceAccess`
- rename the `ColumnAccess` struct to `IndexedTraceAccess`, and change the enum variant here to `TraceAccess`

This will end up with something like the following:
```
   /// TODO: please add a comment here when you update to `NamedTraceAccess`
   Next(NamedTraceAccess),
    /// Represents accessing a column in the execution trace at a particular row offset, e.g. `$main[2]`.
    /// [IndexedTraceAccess] contains the segment index, column index within the segment, and row 
    /// offset to be accessed.
    TraceAccess(IndexedTraceAccess),
```